File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/model/base/QueryManager.java
package com.example.ama.android2_lesson03.model.base;

import android.net.Uri;

import com.google.android.gms.maps.model.LatLng;

public interface QueryManager {
    void prepareGMapUri(String query, OnUriPreparedCallback callback);

    void getFullLocationName(String query, OnFullNamePreparedCallback callback);

    void getFullLocationName(LatLng latLng, OnFullNamePreparedCallback callback);

    void getMyLocation(OnLocationSearchResultCallback callback);

    interface OnUriPreparedCallback {
        void onSuccess(Uri uri);
    }

    interface OnFullNamePreparedCallback {
        void onSuccess(String fullLocationName, LatLng latLng, float zoom);
    }

    interface OnLocationSearchResultCallback {
        void onLocationFound(LatLng latLng);

        void onNotFound();

        void onPermissionRequired(String permission, int requestCode);
    }
}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/repo/base/QueryManager.kt
package com.example.ama.android2_lesson03.repo.base

import android.net.Uri
import com.google.android.gms.maps.model.LatLng

interface QueryManager {
    fun getPreparedUri(isMarkerOnTheMap: Boolean, cameraPosition: LatLng?, zoom: Float?, callback: (uri: Uri) -> Unit)
    fun getFullLocationName(query: String, callback: (fullLocationName: String, latLng: LatLng, zoom: Float) -> Unit)
    fun getFullLocationName(latLng: LatLng, callback: (fullLocationName: String, latLng: LatLng, zoom: Float) -> Unit)
    fun getMyLocation(found: (latLng: LatLng, zoom: Float) -> Unit,
                      notFound: (message: String) -> Unit,
                      permissionRequired: (permission: String, requestCode: Int) -> Unit)
}